catkit2/services/hamamatsu_camera/hamamatsu_camera.py

```diff
@@ -270,8 +264,91 @@ def monitor_temperature(self):
             temperature = self.get_temperature()
             self.temperature.submit_data(np.array([temperature]))
 
+            if temperature > self.critical_temp and self.is_acquiring.get():
+                self.log.warning(f'Camera temperature = {temperature} > {self.critical_temp} degrees.')
+                self.log.warning('Stopping acquisition and start fan.')
+                self.fan_status = True
+                self.cooler_mode = 'on'
+                self.end_acquisition()
+
             self.sleep(0.1)
 
+    @property
+    def cooler_mode(self):
+        """
+        Check the cooling mode.
+
+        Returns
+        -------
+        str :
+            Sensor cooler status 'on', 'off' or 'max'.
+        """
+        coolint = self.cam.prop_getvalue(dcam.DCAM_IDPROP.SENSORCOOLER)
+
+        if coolint == 1.0:
+            return 'off'
+        elif coolint == 2.0:
+            return 'on'
+        else:

```

Comment:
It would be good to make it explicit in the code which int value "max" corresponds to. Otherwise there is no way to know other than reading manufacturer docs.